bigchaindb/models.py

```diff
@@ -74,29 +82,26 @@ def validate(self, bigchain):
                     raise DoubleSpend('input `{}` was already spent'
                                       .format(input_txid))
 
-                input_conditions.append(input_tx.conditions[input_cid])
+                input_condition = input_tx.conditions[input_cid]
+                input_conditions.append(input_condition)
+
                 input_txs.append(input_tx)
-                if input_tx.conditions[input_cid].amount < 1:
-                    raise AmountError('`amount` needs to be greater than zero')
-                input_amount += input_tx.conditions[input_cid].amount
 
             # validate asset id
-            asset_id = Asset.get_asset_id(input_txs)
-            if asset_id != self.asset.id:
+            asset_id = Transaction.get_asset_id(input_txs)
+            if asset_id != self.asset['id']:
                 raise AssetIdMismatch(('The asset id of the input does not'
                                        ' match the asset id of the'
                                        ' transaction'))
 
-            # get the asset creation
-            asset = bigchain.get_asset_by_id(asset_id)
-            # validate the asset
-            asset.validate_asset(amount=input_amount)
             # validate the amounts
-            output_amount = 0
-            for condition in self.conditions:
-                if condition.amount < 1:
-                    raise AmountError('`amount` needs to be greater than zero')
-                output_amount += condition.amount
+            if any(any(condition.amount < 1 for condition in conditions) for
```

Comment:
I don't think that this is needed.
You already validate the amounts of this transaction outputs in https://github.com/bigchaindb/bigchaindb/pull/982/files#diff-9f1ab5b5daf1b3072e2123b6939bfb07R51

And the input conditions where already validated when the input transation was submitted